poc/icart_hacspec.py

```diff
@@ -0,0 +1,57 @@
+from hacspec.speclib import *
+
+prime = 2**384 - 2**128 - 2**96 + 2**32 - 1
+
+felem_t = refine(nat, lambda x: x < prime)
```

Comment:
This changed to change to `refine_t` and `nat_t` to be more consistent using `_t`.

Reply:
ACK!

Reply:
Does this still need to be done?

Reply:
I elected not to do it, since this wouldn't go into the appendix.